src/ngx_http_lua_headers_in.c

```diff
@@ -449,6 +453,9 @@ ngx_http_set_host_header(ngx_http_request_t *r, ngx_http_lua_header_val_t *hv,
         r->headers_in.server = *value;
     }
 
+    var = &r->variables[lmcf->host_index];
+    var->valid = var->not_found = 0;
```

Comment:
Better use 2 assignment statements here for clarity. The whole nginx code base never uses this form of assignment anyway.